### PR TITLE
Support authorization_details in token endpoint

### PR DESCRIPTION
**Pull request by vafeini**
Closes issue  https://github.com/eu-digital-identity-wallet/eudi-lib-jvm-openid4vci-kt/issues/313

### CODE REVIEW

File: src/main/kotlin/eu/europa/ec/eudi/openid4vci/AuthorizeIssuance.kt [hunk 1]
```diff
@@ -26,11 +26,14 @@ import java.time.Instant
...
  * @param state the state which was sent with the
  * authorization request
+ * @param identifiersSentAsAuthDetails The list of the offer's [CredentialConfigurationIdentifier]s that are or will be communicated to
+ * authorization server as detailed authorizations, part of a Rich Authorization Request.
  */
 data class AuthorizationRequestPrepared(
     val authorizationCodeURL: HttpsUrl,
     val pkceVerifier: PKCEVerifier,
     val state: String,
+    val identifiersSentAsAuthDetails: List<CredentialConfigurationIdentifier> = emptyList(),
```
Comment (babisRoutis) [inline src/main/kotlin/eu/europa/ec/eudi/openid4vci/AuthorizeIssuance.kt]: Let's avoid default value of empty list.

In any case this is populated by the lib.

File: src/main/kotlin/eu/europa/ec/eudi/openid4vci/AuthorizeIssuance.kt [hunk 2]
```diff
@@ -113,6 +116,13 @@ sealed interface AuthorizedRequest : java.io.Serializable {
     ) : AuthorizedRequest
 }
 
+sealed interface AuthorizationDetailsInTokenRequest {
```
Comment (babisRoutis) [inline src/main/kotlin/eu/europa/ec/eudi/openid4vci/AuthorizeIssuance.kt]: I would call it something like AccessTokenOption, AsRequested  for DoNotInclude, Limited for Include

File: src/main/kotlin/eu/europa/ec/eudi/openid4vci/internal/http/AuthorizationEndpointClient.kt [hunk 3]
```diff
@@ -196,7 +199,14 @@ internal class AuthorizationEndpointClient(
                 }
             }
             if (credentialsAuthorizationDetails.isNotEmpty()) {
-                authorizationDetails(credentialsAuthorizationDetails.map(::toNimbus))
```
Comment (babisRoutis) [inline src/main/kotlin/eu/europa/ec/eudi/openid4vci/internal/http/AuthorizationEndpointClient.kt:199]: Why method was removed?

File: src/main/kotlin/eu/europa/ec/eudi/openid4vci/internal/http/AuthorizationEndpointClient.kt [hunk 4]
```diff
@@ -251,23 +261,10 @@ internal class AuthorizationEndpointClient(
         else response.body<PushedAuthorizationRequestResponseTO.Failure>()
     }
 
-    private fun toNimbus(
```
Comment (babisRoutis) [inline src/main/kotlin/eu/europa/ec/eudi/openid4vci/internal/http/AuthorizationEndpointClient.kt:254]: Why removed?

File: src/test/kotlin/eu/europa/ec/eudi/openid4vci/CommonAssertions.kt [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+ */
+import eu.europa.ec.eudi.openid4vci.internal.http.TokenEndpointForm
+import io.ktor.client.request.*
+import io.ktor.client.request.forms.*
+import kotlin.test.assertIs
+import kotlin.test.assertNotNull
+import kotlin.test.assertNull
+import kotlin.test.assertTrue
+
+val parPost_ApplyAssertionsAndGetFormData: HttpRequestData.(expectIssuerState: Boolean) -> FormDataContent = { expectIssuerState ->
```
Comment (babisRoutis) [inline src/test/kotlin/eu/europa/ec/eudi/openid4vci/CommonAssertions.kt]: Change the name please to kotlin conventions. Sonar reports this

**Review by babisRoutis — CHANGES_REQUESTED** [review]

File: src/main/kotlin/eu/europa/ec/eudi/openid4vci/internal/http/AuthorizationEndpointClient.kt (hunk 4 above)
Comment (vafeini) [reply src/main/kotlin/eu/europa/ec/eudi/openid4vci/internal/http/AuthorizationEndpointClient.kt:254]: Is moved to [AuthorizationDetailsOps.kt](https://github.com/niscy-eudiw/eudi-lib-jvm-openid4vci-kt/pull/4/files/e5dc09cabd56b0689843b61a1d5d6b799d913e27#diff-971adad0fbb432aad6a708957f833aa4682a2ee2a9b9744994e964bd34dda3d0) along with other utility methods that have to do with how authorization_details are embeded as form parameter

File: src/main/kotlin/eu/europa/ec/eudi/openid4vci/internal/http/AuthorizationEndpointClient.kt (hunk 3 above)
Comment (vafeini) [reply src/main/kotlin/eu/europa/ec/eudi/openid4vci/internal/http/AuthorizationEndpointClient.kt:199]: Is moved to AuthorizationDetailsOps.kt along with other utility methods that have to do with how authorization_details are embeded as form parameter

File: src/main/kotlin/eu/europa/ec/eudi/openid4vci/AuthorizeIssuance.kt (hunk 1 above)
Comment (vafeini) [reply src/main/kotlin/eu/europa/ec/eudi/openid4vci/AuthorizeIssuance.kt]: Done

File: src/main/kotlin/eu/europa/ec/eudi/openid4vci/AuthorizeIssuance.kt (hunk 2 above)
Comment (vafeini) [reply src/main/kotlin/eu/europa/ec/eudi/openid4vci/AuthorizeIssuance.kt]: Done

**Review by babisRoutis — APPROVED** [review]

**Pull request merged**